components/fs/nfs/nfs.mk

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright 2024, UNSW
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# This Makefile snippet builds the NFS component
+# it should be included into your project Makefile
+#
+# NOTES:
+# Generates nfs.elf
+# Requires ${SDDF}/util/util.mk to build the utility library for debug output
+# Requires CONFIG_INCLUDE, NFS_SERVER and NFS_DIRECTORY to be defined
+
+NFS_DIR := $(LIONSOS)/components/fs/nfs
+LWIP := $(SDDF)/network/ipstacks/lwip/src
+MUSL := $(LIONSOS)/dep/musllibc
+LIBNFS := $(LIONSOS)/dep/libnfs
+
+CFLAGS_nfs := \
+	-DNFS_SERVER="\"$(NFS_SERVER)\"" \
+	-DNFS_DIRECTORY="\"$(NFS_DIRECTORY)\"" \
+	-Imusllibc/include \
+	-I$(NFS_DIR)/lwip_include \
+	-I$(LIBNFS)/include \
+	-I$(LWIP)/include \
+	-I$(LWIP)/include/ipv4 \
+
+include $(LWIP)/Filelists.mk
+NFS_NETIFFILES := $(LWIPDIR)/netif/ethernet.c
+NFS_LWIPFILES := $(COREFILES) $(CORE4FILES) $(NFS_NETIFFILES)
+NFS_LWIP_OBJ := $(addprefix nfs/lwip/, $(NFS_LWIPFILES:.c=.o))
+
+NFS_DIRS := nfs $(addprefix nfs/lwip/, api core core/ipv4 netif)
+
+NFS_FILES := nfs.c fd.c op.c posix.c tcp.c
+NFS_OBJ := $(addprefix nfs/, $(NFS_FILES:.c=.o)) $(NFS_LWIP_OBJ)
+
+CHECK_NFS_FLAGS_MD5 := .nfs_cflags-$(shell echo -- $(CFLAGS) $(CFLAGS_nfs) | shasum | sed 's/ *-//')
+
+$(CHECK_NFS_FLAGS_MD5):
+	-rm -f .nfs_cflags-*
+	touch $@
+
+musllibc/lib/libc.a:
```

Comment:
This should be in a separate component.  Suggest passing in the MUSLC location as a variable